### PR TITLE
Firewall modifications and loading fix.

### PR DESCRIPTION
**Pull request by maximelb**
Enabling a `Firewall.update()` function to update an existing firewall.
Fixing loading an existing Firewall where it used port `"0"` instead of `"all"` which caused issue when trying to do a `.load()` followed by an `.update()` since `"0"` is not a valid port in the API.

### CODE REVIEW

File: digitalocean/Firewall.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ class InboundRule(object):
     """
     def __init__(self, protocol="", ports="", sources=""):
         self.protocol = protocol
-        self.ports = ports
+        self.ports = ports if ports != "0" else "all"
```
Comment (koalalorenzo) [inline digitalocean/Firewall.py]: Please add a test for this

File: digitalocean/Firewall.py [hunk 2]
```diff
@@ -195,6 +195,22 @@ def create(self, *args, **kwargs):
 
         return self
 
+    def update(self, *args, **kwargs):
```
Comment (koalalorenzo) [inline digitalocean/Firewall.py]: Please add a test for this

File: digitalocean/Firewall.py [hunk 3]
```diff
@@ -195,6 +195,22 @@ def create(self, *args, **kwargs):
...
+        inbound = jsonpickle.encode(self.inbound_rules, unpicklable=False)
+        outbound = jsonpickle.encode(self.outbound_rules, unpicklable=False)
+        params = {'name': self.name,
```
Comment (koalalorenzo) [inline digitalocean/Firewall.py]: Code is repeated, think about using a function to get the params or refactor to avoid repeated code

**Comment by koalalorenzo** [comment]
Hello! Any update on this?

**Comment by maximelb** [comment]
Hello, no sorry, I was initially hoping the simplicity of the change wouldn’t require tests on my part.

I very much understand the desire for fully packaged contributions to open source projects but to be honest I don’t have much time to learn the test structure and go back and forth on the correctness of the test.

Really is nothing personal but I foresee a possible time sink I can’t afford at the moment. I’m working off of my branch and have found a few other minor issues (great project by the way :) ) so I might just batch those up in my branch and one day go through them and do a few more thorough PRs.

In this case I wasn’t sure if you wanted faster fixes without tests that you would integrate yourself or slower fully packaged PRs.

Cheers

Max

**Pull request closed without merging**